elytron/src/test/java/org/wildfly/extension/elytron/SyslogAuditLogTestCase.java

```diff
@@ -0,0 +1,327 @@
+        badHostUdpOperation = createUdpSyslogOperation(BAD_SERVER_ADDRESS);
+    }
+
+    /**
+     * Initializes the syslog server for all tests
+     */
+    @BeforeClass
+    public static void initServer() throws Exception {
+        udpServer = SimpleSyslogServer.createUdp(UDP_PORT);
+    }
+
+    /**
+     * Resets the operations after each test
+     */
+    @After
+    public void shutdownServer() {
+        udpOperation = null;
+        badHostUdpOperation = null;
+    }
+
+    /**
+     * Closes the syslog server after all tests
+     */
+    @AfterClass
+    public static void closeServer() {
+        udpServer.close();
+    }
+
+    /**
+     * Sets the syslog format to RFC3164 and verifies the Elytron AuditEndpoint is created with RFC3164
+     * by verifying the connection message
+     */
+    @Test
+    public void testRfc3164SyslogFormat() throws Exception {
+        udpOperation.get(ElytronDescriptionConstants.SYSLOG_FORMAT).set(RFC3164_STRING);
+        assertSuccess(services.executeOperation(udpOperation));
+        verifyUdpMessage(RFC3164_SYSLOG_MESSAGE);
+    }
+
+
+    /**
+     * Sets the syslog format to RFC5424 and verifies the Elytron AuditEndpoint is created with RFC5424
+     * by verifying the connection message
+     */
+    @Test
+    public void testRfc5424SyslogFormat() throws Exception {
+        udpOperation.get(ElytronDescriptionConstants.SYSLOG_FORMAT).set(RFC5424_STRING);
+        assertSuccess(services.executeOperation(udpOperation));
+        verifyUdpMessage(RFC5424_SYSLOG_MESSAGE);
+    }
+
+
+    /**
+     * Doesn't set a syslog and verifies the Elytron AuditEndpoint is created with the default value
+     * of RFC5424 by verifying the connection message
+     */
+    @Test
+    public void testDefaultSyslogFormat() throws Exception {
+        assertSuccess(services.executeOperation(udpOperation));
+        verifyUdpMessage(RFC5424_SYSLOG_MESSAGE);
+    }
+
+    /**
+     * Tests that the syslog configuration operation fails if an invalid RFC format is used
+     */
+    @Test
+    public void testWrongSyslogFormat() {
+        udpOperation.get(ElytronDescriptionConstants.SYSLOG_FORMAT).set(BAD_RFC_STRING);
+        ModelNode response = services.executeOperation(udpOperation);
+        assertCorrectError(response, "No enum constant org.jboss.logmanager.handlers.SyslogHandler.SyslogType.RFC1");
+        assertFailed(response);
+    }
+
+    /**
+     * Tests that the syslog configuration operation fails if an invalid icmp-error-reconnect-attempts number is used
+     */
+    @Test
+    public void testBadIntegerReconnectAttempts() {
+        udpOperation.get(ElytronDescriptionConstants.ICMP_ERROR_RECONNECT_ATTEMPTS).set(BAD_RECONNECT_NUMBER);
+        ModelNode response = services.executeOperation(udpOperation);
+        assertCorrectError(response, BAD_RECONNECT_NUMBER + " is an invalid value for parameter icmp-error-reconnect-attempts. A minimum value of -1 is required");
```

Comment:
Same as above. We should use the message ID or assume a failure is enough.

Reply:
Updated